compiler/hash-parser/src/parser/expression.rs

```diff
@@ -388,6 +388,8 @@ impl<'stream, 'resolver> AstGen<'stream, 'resolver> {
                     // if the operator is a non-functional, (e.g. as) we need to perform a different
                     // conversion where we transform the AstNode into a
                     // different
+                    // @@Todo: Use `ExpressionKind::TypeExpr` rather than relying on
+                    // `ExpressionKind::CastExpr`...
```

Comment:
Maybe this would be a bit annoying to do ?

Reply:
Yeah, maybe it should just be left like this?

Reply:
Yes, ill remove this comment and then merge